### PR TITLE
All-in on graphmos

### PR DESCRIPTION
**Pull request by Putnam3145**
It's all graphs all the time baby oh yeah. Seriously, though, there's a lot here and I'm not sure what's all going to go in this. "Ideally" we should allow basically any arbitrary type of atmos this way, including e.g. simulating pipenets, but at that point I'd rather just start working on SS14 hahaha.

### CODE REVIEW

File: src/turfs.rs [hunk 1]
```diff
@@ -167,50 +163,57 @@ struct ThermalInfo {
...
-	pub fn update_adjacencies(&mut self, idx: TurfID, adjacent_list: List) -> Result<(), Runtime> {
-		if let Some(&this_index) = self.map.get(&idx) {
-			self.remove_adjacencies(idx);
+	pub fn turf_id_map(&self) -> TurfGraphMap {
+		self.map
+			.lock()
+			.get_or_insert_with(|| {
+				self.graph
+					.raw_nodes()
+					.iter()
```
Comment (jupyterkat) [inline src/turfs.rs]: this can be replaced with par_iter(). but the performance increase is argueable

File: src/turfs/processing.rs [hunk 2]
```diff
@@ -99,6 +99,59 @@ fn _finish_process_turfs() {
 	}
```
Comment (jupyterkat) [inline src/turfs/processing.rs:97]: forgot to replace above callbacks with graph rebuilds

File: src/turfs/processing.rs [hunk 3]
```diff
@@ -172,172 +224,207 @@ fn _process_turf_start() -> Result<(), String> {
...
 	index: NodeIndex<usize>,
 	mixture: &TurfMixture,
 	all_mixtures: &[RwLock<Mixture>],
-	arena: &RwLockReadGuard<TurfGases>,
+	arena: &TurfGases,
 ) -> bool {
-	arena.adjacent_infos(index).next().is_some()
-		&& mixture.enabled()
+	mixture.enabled()
+		&& arena.adjacent_infos(index).next().is_some()
```
Comment (jupyterkat) [inline src/turfs/processing.rs]: singletons are already filtered out??

Comment (Putnam3145) [reply src/turfs/processing.rs]: this is an issue but it's a separate one, one sec

Comment (Putnam3145) [reply src/turfs/processing.rs]: wait no i undid having space turfs have no outgoings.

**Pull request merged**